ptsemseg/models/linknet.py

```diff
@@ -23,41 +33,36 @@ def __init__(self, feature_scale=4, n_classes=21, is_deconv=True, in_channels=3,
                                                padding=3, stride=2, bias=False)
         self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
 
-        block = residualBlock
-        self.encoder1 = self._make_layer(block, filters[0], self.layers[0])
-        self.encoder2 = self._make_layer(block, filters[1], self.layers[1], stride=2)
-        self.encoder3 = self._make_layer(block, filters[2], self.layers[2], stride=2)
-        self.encoder4 = self._make_layer(block, filters[3], self.layers[3], stride=2)
+        block = Resnets[resnet]['block']
+        self.encoder1 = self._make_layer(block, filters[0], layers[0])
+        self.encoder2 = self._make_layer(block, filters[1], layers[1], stride=2)
+        self.encoder3 = self._make_layer(block, filters[2], layers[2], stride=2)
+        self.encoder4 = self._make_layer(block, filters[3], layers[3], stride=2)
         self.avgpool = nn.AvgPool2d(7)
 
 
         # Decoder
-        self.decoder4 = linknetUp(filters[3], filters[2])
-        self.decoder4 = linknetUp(filters[2], filters[1])
-        self.decoder4 = linknetUp(filters[1], filters[0])
-        self.decoder4 = linknetUp(filters[0], filters[0])
+        self.decoder4 = linknetUp(filters[3]*expansion, filters[2]*expansion)
+        self.decoder3 = linknetUp(filters[2]*expansion, filters[1]*expansion)
+        self.decoder2 = linknetUp(filters[1]*expansion, filters[0]*expansion)
+        self.decoder1 = linknetUp(filters[0]*expansion, filters[0])
 
         # Final Classifier
-        self.finaldeconvbnrelu1 = nn.Sequential(nn.ConvTranspose2d(filters[0], 32/feature_scale, 3, 2, 1),
-                                      nn.BatchNorm2d(32/feature_scale),
-                                      nn.ReLU(inplace=True),)
+        self.finaldeconvbnrelu1 = deconv2DBatchNormRelu(filters[0], 32/feature_scale, 2, 2, 0)
         self.finalconvbnrelu2 = conv2DBatchNormRelu(in_channels=32/feature_scale, k_size=3, n_filters=32/feature_scale, padding=1, stride=1)
-        self.finalconv3 = nn.Conv2d(32/feature_scale, n_classes, 2, 2, 0)
+        self.finalconv3 = nn.Conv2d(int(32/feature_scale), int(n_classes), 3, 1, 1)
```

Comment:
Yes, that is correct. in short it works fine for dimensions divisible by 16 since there are 4 times down-sampling and 4 times up-sampling of a factor of 2.